### DISCUSSION TITLE
panic: division by zero when running the ckks/boostrapping example

### DISCUSSION THREAD

**Discussion by ghost** [tuneinsight/lattigo#374] [Q&A] [answered]
I'm new to Lattigo，and try to learn about the ckks-bootstrapping demo in lattigo/examples/ckks/bootstrapping.
I change nothing and run the main.go file of this demo, but get the error "panic : division by zero", how should I fix this？
Thanks!

This is the output detail:

`Bootstrapping depth (bits): 822
panic: division by zero

goroutine 1 [running]:
math/big.nat.div(0x0, 0x0, 0x0, 0xc00164a250, 0x1, 0x1, 0xc00164a1a0, 0x1, 0x1, 0xc0017bba60, ...)
        /usr/lib/go-1.13/src/math/big/nat.go:648 +0x481
math/big.(*Int).QuoRem(0xc0001dbf70, 0xc001649160, 0xc0017be0c0, 0xc0001dc0a0, 0x0, 0xc0017bba68)
        /usr/lib/go-1.13/src/math/big/int.go:239 +0xbf
math/big.(*Int).Mod(0xc0001dc0a0, 0xc001649160, 0xc0017be0c0, 0xc0001dc0a0)
        /usr/lib/go-1.13/src/math/big/int.go:270 +0x10a
github.com/tuneinsight/lattigo/v4/utils.GetFactors(0xc0001dc218, 0x0, 0x0, 0x0)
        /home/user/lattigo/utils/factorization.go:66 +0x3e9
github.com/tuneinsight/lattigo/v4/ring.PrimitiveRoot(0xfffffffff6a0001, 0x0, 0x0, 0x0, 0x17, 0xc00018f080, 0x6, 0x6, 0x0, 0x0)
        /home/user/lattigo/ring/subring.go:165 +0xaa
github.com/tuneinsight/lattigo/v4/ring.(*SubRing).generateNTTConstants(0xc0002d47e0, 0x0, 0x0)
        /home/user/lattigo/ring/subring.go:120 +0xe9
github.com/tuneinsight/lattigo/v4/ring.(*Ring).generateNTTConstants(0xc0001c00a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x1fffffffff500001)
        /home/user/lattigo/ring/ring.go:326 +0x11e
github.com/tuneinsight/lattigo/v4/ring.NewRingWithCustomNTT(0x10000, 0xc00029a000, 0x1a, 0x1a, 0x5cbde0, 0x20000, 0x72d9e0, 0x7ff4a48ae6d0, 0x0)
        /home/user/lattigo/ring/ring.go:287 +0x4b8
github.com/tuneinsight/lattigo/v4/ring.NewRingFromType(0x10000, 0xc00029a000, 0x1a, 0x1a, 0x0, 0x5826a0, 0x1, 0xc0001d0160)
        /home/user/lattigo/ring/ring.go:240 +0x141
github.com/tuneinsight/lattigo/v4/rlwe.(*Parameters).initRings(0xc0001dc5c0, 0xc0001d0140, 0x20)
        /home/user/lattigo/rlwe/params.go:851 +0x71
github.com/tuneinsight/lattigo/v4/rlwe.NewParameters(0x10, 0xc000298000, 0x1a, 0x20, 0xc0001d0140, 0x4, 0x4, 0x0, 0xc0, 0x400999999999999a, ...)
        /home/user/lattigo/rlwe/params.go:136 +0x373
github.com/tuneinsight/lattigo/v4/rlwe.NewParametersFromLiteral(0x10, 0xc000298000, 0x1a, 0x20, 0xc0001d0140, 0x4, 0x4, 0x0, 0x0, 0x0, ...)
        /home/user/lattigo/rlwe/params.go:175 +0x5b4
github.com/tuneinsight/lattigo/v4/ckks.NewParametersFromLiteral(0x10, 0xc000298000, 0x1a, 0x20, 0xc0001d0140, 0x4, 0x4, 0x0, 0x0, 0x0, ...)
        /home/user/lattigo/ckks/params.go:345 +0xd7
main.main()
        /home/user/lattigo/examples/ckks/bootstrapping/main.go:77 +0x442
exit status 2`

**Comment by ghost** [comment]
This is the btpParams and ckksParamsLit:
btpParams: {{1 0 0 13 [1 1 1] true 0 false 1} {21 60 0 8 16 30 3 0} {0 0 0 25 [1 1 1 1] true 0 false 1} 1 32}
ckksParamsLit: {16 [36028797019488257 1099512938497 1099510054913 1099507695617 1099515691009 1099506515969 1099516870657 1099504549889 1099503894529 1099503370241 1099502714881 549754109953 549753978881 549753716737 1152921504606584833 1152921504614055937 1152921504598720513 1152921504615628801 1152921504597016577 1152921504616808449 1152921504595968001 1152921504618381313 72057594038321153 72057594036879361 72057594035306497 72057594040680449] [2305843009211596801 2305843009210023937 2305843009208713217 2305843009202159617] [] [] 0 0 192 Standard 15 40}

**Comment by ghost** [comment]
Although I don't know what happens exactly, but after updating go version from 1.13 to 1.20, the demo can run successfully.

**Comment by Pro7ech** [comment, marked as answer]
Hi @ZRcrypto, indeed the `go.mod` specifies that the minimum Go version to run Lattigo is 1.14.

**Reply by ghost** [reply]
Okay， thank you very much.
